src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+    const std::vector<std::pair<T, T>>& additional_bin_edges) const {
+  if constexpr (std::is_same_v<
+                    T,
+                    pmr_string>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+    Fail("Cannot split_at_bin_bounds() on string histogram");
+  }
+
+  /**
+   * Create vector with pairs for each split.
+   * For a lower bin edge e, the new histogram will have to be split at previous_value(e) and e.
+   * For an upper bin edge e, the new histogram will have to be split at e and next_value(e).
+   *
+   * We can safely ignore duplicate splits, so we use a set first and then copy the splits to a vector.
+   * Also, we will create splits in gaps for now, but we will not create any actual bins for them later on.
+   */
+  const auto current_bin_count = bin_count();
+
+  std::unordered_set<std::pair<T, T>, boost::hash<std::pair<T, T>>> split_set;
+
+  for (auto bin_id = BinID{0}; bin_id < current_bin_count; bin_id++) {
+    const auto bin_min = bin_minimum(bin_id);
+    const auto bin_max = bin_maximum(bin_id);
+    if constexpr (std::is_arithmetic_v<
+                      T>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+      split_set.insert(std::make_pair(_domain.previous_value(bin_min), bin_min));
+      split_set.insert(std::make_pair(bin_max, _domain.next_value(bin_max)));
+    } else {
+      // TODO(tim): turn into compile-time error
+      Fail("Not supported for strings.");
+    }
+  }
+
+  for (const auto& edge_pair : additional_bin_edges) {
+    if constexpr (std::is_arithmetic_v<
+                      T>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+      split_set.insert(std::make_pair(_domain.previous_value(edge_pair.first), edge_pair.first));
+      split_set.insert(std::make_pair(edge_pair.second, _domain.next_value(edge_pair.second)));
+    } else {
+      // TODO(tim): turn into compile-time error
+      Fail("Not supported for strings.");
+    }
+  }
+
+  /**
+   * Split pairs into single values.
+   * These are the new bin edges, and we have to sort them.
+   */
+  std::vector<T> all_edges(split_set.size() * 2);
+  size_t edge_idx = 0;
+  for (const auto& split_pair : split_set) {
+    all_edges[edge_idx] = split_pair.first;
+    all_edges[edge_idx + 1] = split_pair.second;
+    edge_idx += 2;
+  }
+
+  std::sort(all_edges.begin(), all_edges.end());
+
+  /**
+   * Remove the first and last value.
+   * These are the values introduced in the first step for the first and last split.
+   * The lower edge of bin 0 basically added the upper edge of bin -1,
+   * and the upper edge of the last bin basically added the lower edge of the bin after the last bin.
+   * Both values are obviously not needed.
+   */
+  all_edges.erase(all_edges.begin());
+  all_edges.pop_back();
+
+  // We do not resize the vectors because we might not need all the slots because bins can be empty.
```

Comment:
Rephrase?

Reply:
ping

Reply:
Should hopefully be addressed by https://github.com/hyrise/hyrise/pull/1489#discussion_r291065394

Reply:
That comment made no sense to me either... I removed it